src/main/scala/scalafix/sbt/ScalafixPlugin.scala

```diff
@@ -226,18 +239,78 @@ object ScalafixPlugin extends AutoPlugin {
 
   private def runArgs(
       interface: ScalafixInterface,
-      logger: Logger
+      streams: TaskStreams
   ): Unit = {
     val paths = interface.args.collect { case Arg.Paths(paths) => paths }.flatten
     if (paths.nonEmpty) {
       if (paths.lengthCompare(1) > 0) {
-        logger.info(s"Running scalafix on ${paths.size} Scala sources")
+        streams.log.info(s"Running scalafix on ${paths.size} Scala sources")
+      }
+
+      val cacheKeyArgs = interface.args.collect {
+        case cacheKey: Arg.CacheKey => cacheKey
+      }
+      val pathsFilesInfo = FilesInfo
+        .lastModified(paths.map(_.toFile).toSet)
+        .asInstanceOf[FilesInfo[ModifiedFileInfo]]
+
+      // used to signal that one of the argument cannot be reliably stamped
+      object StampingImpossible extends RuntimeException with NoStackTrace
+
+      implicit val stamper = new CacheKeysStamper {
+        override protected def stamp: Arg.CacheKey => Unit = {
```

Comment:
The overall stamp is the concatenation of each arg stamp, taken in the order they were set, which depends on the flow of the `scalafix` `InputTask` implementation (so relatively stable). As we control this order (as opposed to the user, which can only affect the order of the `ParsedArgs`), it's not necessary to append the type/name of each arg to disambuiguate 2 args that would have the same value.